kube-api-core/src/kube_api/core/auth.clj

```diff
@@ -148,6 +148,73 @@
                 (assoc-in request [:headers "Authorization"] (str "Bearer " new-token))))]
       (update client-opts :middleware (fnil conj []) (retry-middleware gen-new-request)))))
 
+; https://github.com/kubernetes/client-go/blob/v0.21.0/plugin/pkg/client/auth/gcp/gcp.go
+(defmethod inject-client-auth :gcp-provider [client-opts user]
+  (let [{:keys [access-token scopes cmd-args cmd-path expiry token-key expiry-key time-fmt]
+         :or   {scopes     "https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/userinfo.email"
+                token-key  "{.access_token}"
+                expiry-key "{.token_expiry}"
+                time-fmt   "2006-01-02T15:04:05.999999999Z07:00"}}
```

Comment:
My GKE config time-expiry is in ISO8601/RFC3339 format. `java.time.Instant/parse` works just fine for the aforementioned formats. Is there a use for RFC3339Nano? Quick readings seem to suggest that it is a fringe format which exists only in the Golang universe.

Reply:
Interesting. I was using RFC3339Nano because I read it was the default format if time-fmt is omitted: https://github.com/kubernetes/client-go/blob/v0.21.0/plugin/pkg/client/auth/gcp/gcp.go#L104-L106

But it sounds like your format is RFC3339 (not nano) by default and your kubeconfig doesn't specify "time-fmt" ? I suppose to be safe I should just make parse ISO to any level of precision, but was finding that `Instant/parse` doesn't work on the nano precision out of the box.

Reply:
Yeah I do not have time-fmt in my kube config and my expiry timestamps are in RFC3339. This is the Golang client for K8. I believe this behaviour is Golang specific. There is no mention of this format in other client libraries (at least for Python)